spring-cloud-app-broker-deployer-cloudfoundry/src/main/java/org/springframework/cloud/appbroker/deployer/cloudfoundry/CloudFoundryAppDeployer.java

```diff
@@ -580,14 +585,50 @@ private Mono<String> upgradeApplicationIfRequired(UpdateApplicationRequest reque
 		return getPackageForApplication(applicationId);
 	}
 
-	private Mono<String> updateEnvironment(UpdateApplicationRequest request, String applicationId) {
-		final Map<String, Object> environmentVariables =
-			getApplicationEnvironment(request.getProperties(), request.getEnvironment(),
-				request.getServiceInstanceId());
-		return updateApplicationEnvironment(applicationId, environmentVariables, request.getProperties())
+	private Mono<String> updateApplication(UpdateApplicationRequest request, String applicationId) {
+		final Map<String, Object> environmentVariables = getApplicationEnvironment(request.getProperties(),
+			request.getEnvironment(), request.getServiceInstanceId());
+		Map<String, String> properties = request.getProperties();
+
+		return updateStackIfPresent(request, applicationId)
+			.then(this.client.applicationsV2()
+				.update(builder()
+					.applicationId(applicationId)
+					.instances(instances(properties))
+					.diskQuota(diskQuota(properties))
+					.memory(memory(properties))
+					.putAllEnvironmentJsons(environmentVariables)
+					.build())
+				.doOnRequest(l -> LOG.debug("Updating environment. applicationId={}", applicationId))
+				.doOnSuccess(response -> {
+					LOG.info("Success updating environment. applicationId={}", applicationId);
+					LOG.debug(RESPONSE_LOG_TEMPLATE, response);
+				})
+				.doOnError(e -> LOG.error(String.format("Error updating environment. applicationId=%s, " +
+					ERROR_LOG_TEMPLATE, applicationId, e.getMessage()), e)))
 			.thenReturn(applicationId);
 	}
 
+	private Mono<String> updateStackIfPresent(UpdateApplicationRequest request, String applicationId) {
+		if (request.getProperties().containsKey("upgrade") &&
```

Comment:
Could maybe be a guard clause rather than a full method length `if` block

Reply:
Done, thanks!